File: DataStructuresInJava/src/main/java/implementations/generics/MyMultipleGeneric.java
package implementations.generics;

import com.sun.tools.javac.util.Pair;

public class MyMultipleGeneric<T, U> {
    private T valueT;
    private U valueU;

    public MyMultipleGeneric(T valueT, U valueU)
    {
        this.valueT = valueT;
        this.valueU = valueU;
    }

    public Pair<T, U> GetPair()
    {
        return new Pair<T, U>(valueT, valueU);
    }
}

File: DataStructuresInCSharp/DataStructuresInCSharp/Implementations/0_Generics/MyMultipleGeneric.cs
using System.Collections.Generic;

namespace DataStructuresInCSharp.Implementations._0_Generics
{
    public class MyMultipleGeneric<T, U>
    {
        private readonly T valueT;
        private readonly U valueU;

        public MyMultipleGeneric(T valueT, U valueU)
        {
            this.valueT = valueT;
            this.valueU = valueU;
        }

        public KeyValuePair<T, U> GetPair()
        {
            return new KeyValuePair<T, U>(valueT, valueU);
        }
    }
}
